### PR TITLE
Mem cache with resize

### PR DESCRIPTION
**Pull request by vinnamkim**
### Summary

- Add capability to resize image before mem caching

### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have added e2e tests for validation.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2023 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -269,6 +269,10 @@ max-returns = 10
 "src/otx/**/*.py" = [
     "ERA001",
 ]
+# See https://github.com/openvinotoolkit/training_extensions/actions/runs/7109500350/job/19354528819?pr=2700
```
Comment (sungmanc) [inline pyproject.toml]: It seems that there is a mis-rebase, could you check? this line already made from https://github.com/openvinotoolkit/training_extensions/pull/2700/files#diff-50c86b7ed8ac2cf95bd48334961bf0530cdc77b5a56f852c5c61b89d735fd711R272

Comment (vinnamkim) [reply pyproject.toml]: Fixed in https://github.com/openvinotoolkit/training_extensions/pull/2702/commits/03248baeaa957fddbeb21d631d71b9e651f73429

**Review by sungmanc — DISMISSED** [review]

**Review by jaegukhyun — DISMISSED** [review]
Segmentation is enabled in v2. Will segmentation be treated in this PR or another PR?

**Review by harimkang — DISMISSED** [review]

File: src/otx/core/data/dataset/detection.py [hunk 2]
```diff
@@ -9,26 +9,23 @@
...
 
-    def __init__(self, dm_subset: DatasetSubset, transforms: Transforms) -> None:
-        super().__init__(dm_subset, transforms)
-
     def _get_item_impl(self, index: int) -> DetDataEntity | None:
         item = self.dm_subset.get(id=self.ids[index], subset=self.dm_subset.name)
         img = item.media_as(Image)
         img_data = self._get_img_data(img)
-        img_shape = img.size
+        img_shape = img_data.shape[:2]
```
Comment (jaegukhyun) [inline src/otx/core/data/dataset/detection.py]: Just for double check, doesn't  OTXSegmentationDataset need this change?

Comment (vinnamkim) [reply src/otx/core/data/dataset/detection.py]: Fixed in https://github.com/openvinotoolkit/training_extensions/pull/2702/commits/cf65073304632490a2ab47e916182e53ad63932d

**Review by jaegukhyun — APPROVED** [review]

**Review by harimkang — APPROVED** [review]

**Review by sungmanc — APPROVED** [review]

**Pull request merged**